tests/unit/amo/components/TestCollectionManager.js

```diff
@@ -625,8 +641,93 @@ describe(__filename, () => {
       root, propName: 'onSuggestionSelected',
     });
     selectSuggestion(suggestion);
-    // TODO: test onSuggestionSelected
-    // https://github.com/mozilla/addons-frontend/issues/4590
+
+    sinon.assert.calledWith(dispatchSpy, addAddonToCollection({
+      addonId: suggestion.addonId,
+      collectionId: collection.id,
+      collectionSlug: collection.slug,
+      editing: true,
+      errorHandlerId: errorHandler.id,
+      page,
+      userId: signedInUserId,
+    }));
+  });
+
+  it('dispatches addAddonToCollection with a default page of 1 when selecting an add-on', () => {
+    const errorHandler = createStubErrorHandler();
+
+    const collection = createInternalCollection({
+      detail: createFakeCollectionDetail({ authorUsername: signedInUsername }),
+    });
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const root = render({ collection, errorHandler, page: undefined });
+
+    const suggestion = createInternalSuggestion(
+      createFakeAutocompleteResult({ name: 'uBlock Origin' })
+    );
+    const selectSuggestion = simulateAutoSearchCallback({
+      root, propName: 'onSuggestionSelected',
+    });
+    selectSuggestion(suggestion);
+
+    sinon.assert.calledWith(dispatchSpy, addAddonToCollection({
+      addonId: suggestion.addonId,
+      collectionId: collection.id,
+      collectionSlug: collection.slug,
+      editing: true,
+      errorHandlerId: errorHandler.id,
+      page: 1,
+      userId: signedInUserId,
+    }));
+  });
+
+  it('sets the addonAddedStatus state to pending when selecting an add-on', () => {
+    const root = render({});
+
+    const state = root.state();
+    expect(state.addonAddedStatus).toEqual(null);
+
+    const suggestion = createInternalSuggestion(
+      createFakeAutocompleteResult({ name: 'uBlock Origin' })
+    );
+    const selectSuggestion = simulateAutoSearchCallback({
+      root, propName: 'onSuggestionSelected',
+    });
+    selectSuggestion(suggestion);
+
+    const newState = root.state();
+    expect(newState.addonAddedStatus).toEqual(ADDON_ADDED_STATUS_PENDING);
+  });
+
+  it('displays a notification after an add-on has been added', () => {
+    const root = render({});
+
+    expect(root.find(Notice)).toHaveLength(0);
+
+    root.setProps({ hasAddonBeenAdded: true });
+
+    expect(root.find(Notice)).toHaveLength(1);
+    expect(root.find(Notice).children()).toHaveText('Added to collection');
+  });
+
+  it('removes the notification after a new add-on has been selected', () => {
+    const root = render({});
+
+    expect(root.find(Notice)).toHaveLength(0);
+
+    root.setProps({ hasAddonBeenAdded: true });
```

Comment:
Can you add a comment to say that we set this prop to `true` in order to simulate an add-on already added previously?